File: JavaApp/src/main/java/com/charlezz/javaapp/feature/remote/PostActivity.java
package com.charlezz.javaapp.feature.remote;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.charlezz.javaapp.databinding.ActivityPostBinding;

import javax.inject.Inject;

import dagger.android.support.DaggerAppCompatActivity;

public class PostActivity extends DaggerAppCompatActivity {

    @Inject
    ActivityPostBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        getSupportFragmentManager().beginTransaction()
                .replace(R.id.container, new PostFragment())
                .commit();
    }
}

File: KotlinApp/src/main/java/com/charlezz/finalarchitecture/feature/remote/PostActivity.kt
package com.charlezz.finalarchitecture.feature.remote

import android.os.Bundle
import com.charlezz.finalarchitecture.R
import com.charlezz.finalarchitecture.databinding.ActivityPostBinding
import com.charlezz.finalarchitecture.extension.replaceFragment
import dagger.android.support.DaggerAppCompatActivity
import javax.inject.Inject

class PostActivity : DaggerAppCompatActivity() {

    @Inject
    lateinit var binding: ActivityPostBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if(savedInstanceState==null){
            replaceFragment(R.id.container, PostFragment())
        }
    }
}
